services/nexus/nexus.tester.js

```diff
@@ -281,3 +281,13 @@ t.create('Nexus 3 - repository version with query')
     label: 'nexus',
     message: isVersion,
   })
+
+t.create('Nexus 3 - search release version without snapshots')
+  .get(
+    // Limit the version from above, so that any later artifacts don't break this test.
+    '/r/org.pentaho.adaptive/daemon.json?server=https://nexus.pentaho.org&nexusVersion=3&maven.version=<=9.0.0.0-20180423.155306-4'
```

Comment:
Using `maven.version` like this will be interpreted as a Shields query parameter and will not be forwarded to Nexus. You should use something like this instead:
```
    `/r/org.pentaho.adaptive/daemon.json?server=https://nexus.pentaho.org&nexusVersion=3&queryOpt=${encodeURIComponent(
      ':maven.version=<=9.0.0.0-20180423.155306-4'
    )}`
```

However:
* the `queryParamSchema` pattern may need some tweaking.
* I don't think `maven.version` is a valid parameter (https://nexus.pentaho.org/service/rest/v1/search?group=org.pentaho.adaptive&name=daemon&prerelease=false&maven.version=%3C=9.0.0.0-20180423.155306-4 returns no results). `version` or `maven.baseVersion` may be what we're looking for?

Reply:
You are correct on all counts.
1. `maven.version` is not a valid Nexus 3 parameter. I've changed it to `maven.baseVersion`.
1. Since I had used the last available version as the limit, the test still worked, even though the parameter was ignored. I've lowered the limit, so the most recent versions are not considered (to make sure the limit works), but the latest filtered component is still a snapshot: https://nexus.pentaho.org/service/rest/v1/search?group=org.pentaho.adaptive&name=daemon&sort=version&maven.baseVersion=%3C8.1.0.1
1. I've passed `maven.baseVersion` via `queryOpt`.
1. I've relaxed the value part of the `queryOpt` regex - anything is allowed, as long as it doesn't contain a colon.